serverless/aws/protomaps-template.yaml

```diff
@@ -0,0 +1,187 @@
+    Properties:
+      Action: lambda:InvokeFunctionUrl
+      FunctionName: !Ref ProtomapsLambdaFunction
+      Principal: '*'
+      FunctionUrlAuthType: NONE
+
+# ##########################################################################
+# #  CloudFront::Distribution                                              #
+# ##########################################################################
+
+  CloudFrontDistribution:
+    Type: 'AWS::CloudFront::Distribution'
+    Properties:
+      DistributionConfig:
+        Origins:
+          - Id: ProtomapsLambdaOrigin
+            DomainName: !Select [2, !Split ["/", !GetAtt ProtomapsLambdaFunctionUrl.FunctionUrl]]
+            CustomOriginConfig:
+              OriginProtocolPolicy: https-only
+        DefaultCacheBehavior:
+          TargetOriginId: ProtomapsLambdaOrigin
+          ViewerProtocolPolicy: redirect-to-https
+          CachePolicyId: !Ref CachePolicyId
+          ResponseHeadersPolicyId: !Ref ResponseHeadersPolicyId
+        Enabled: true
+        HttpVersion: http2and3
+        Comment: "Protomaps CloudFront Distribution"
+        PriceClass: PriceClass_All  # Change this to save cost and distribute to fewer countries. Check https://aws.amazon.com/cloudfront/pricing/
+
+# ##########################################################################
+# #  CloudFront::CachePolicy                                               #
+# ##########################################################################
+
+  CachePolicyId:
+    Type: 'AWS::CloudFront::CachePolicy'
+    Properties:
+      CachePolicyConfig:
+        Name: 'CachingOptimized'
+        DefaultTTL: 86400
+        MaxTTL: 31536000
+        MinTTL: 0
+        ParametersInCacheKeyAndForwardedToOrigin:
+          EnableAcceptEncodingBrotli: true
+          EnableAcceptEncodingGzip: true
+          HeadersConfig:
+            HeaderBehavior: none
+          CookiesConfig:
+            CookieBehavior: none
+          QueryStringsConfig:
+            QueryStringBehavior: none
+
+  ResponseHeadersPolicyId:
+    Type: 'AWS::CloudFront::ResponseHeadersPolicy'
+    Properties:
+      ResponseHeadersPolicyConfig:
+        Name: 'protomaps-cors'
+        CorsConfig:
+          AccessControlAllowOrigins:
+            Items: 
+              - 'https://example.com'  # Replace with your allowed origin
+          AccessControlAllowHeaders:
+            Items: 
+              - '*'
+          AccessControlAllowMethods:
+            Items:
+              - GET
+              - POST
+              - OPTIONS
+          AccessControlAllowCredentials: false  # Set to true if you want to include credentials
+          OriginOverride: true
+        Comment: 'CORS policy for Protomaps'
+    DeletionPolicy: Delete
+
+Outputs:
+  BucketNameOutput:
+    Description: 'URL of the S3 bucket'
+    Value: !Sub "https://s3.console.aws.amazon.com/s3/buckets/${BucketName}"
+    Export:
+      Name: !Sub "${AWS::StackName}-S3BucketURL"
+
+  LambdaFunctionUrl:
```

Comment:
I think we can remove this block because the LambdaFunctionUrl is an implementation detail.